src/server/injected/componentUtils.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+type ParsedComponentAttribute = {
+  jsonPath: string[],
+  op: '='|'*=',
+  value: any,
+  caseSensetive: boolean,
+};
+type ParsedComponentSelector = {
+  name: string,
+  attributes: ParsedComponentAttribute[],
+};
+
+export function checkComponentAttribute(obj: any, attr: ParsedComponentAttribute) {
+  for (const token of attr.jsonPath) {
+    if (obj)
+      obj = obj[token];
+  }
+  if (typeof obj !== typeof attr.value)
+    return false;
+  if (typeof obj === 'string') {
+    if (attr.op === '=')
+      return attr.caseSensetive ? obj === attr.value : obj.toUpperCase() === attr.value.toUpperCase();
+    if (attr.op === '*=')
+      return attr.caseSensetive ? obj.includes(attr.value) : obj.toUpperCase().includes(attr.value);
+  }
+  if (attr.op === '=')
+    return obj === attr.value;
+  return false;
+}
+
+export function parseComponentSelector(selector: string): ParsedComponentSelector {
+  let wp = 0;
+  let EOL = selector.length === 0;
+  let next = selector.length ? selector[0] : '';
+  const eat1 = () => {
+    const result = next;
```

Comment:
I think we should handle `\"`-style escapes here.

Reply:
done. not here though - in quoted string.